WordPress/Classes/Stores/PluginStore.swift

```diff
@@ -142,15 +159,32 @@ class PluginStore: QueryStore<PluginStoreState, PluginQuery> {
     }
 
     func processQueries() {
+        // Fetching installed Plugins.
         let sitesToFetch = activeQueries
+            .filter {
+                if case .all = $0 { return true }
+                else { return false }
+            }
             .flatMap { $0.site }
             .unique
             .filter { shouldFetch(site: $0) }
 
-        sitesToFetch.forEach { (site) in
-            fetchPlugins(site: site)
-        }
+        sitesToFetch
+            .forEach { fetchPlugins(site: $0) }
 
+        // Fetching featured plugins.
+        let featuredToFetch  = activeQueries
+            .filter {
+                if case .featured = $0 { return true }
+                else { return false }
+            }
+            .flatMap { $0.site }
+            .unique
+            .filter { state.featuredPluginsSlugs[$0, default: []].isEmpty }
+
+        featuredToFetch.forEach { fetchFeaturedPlugins(site: $0) }
```

Comment:
Related to my previous comment, the featured endpoint will always return the same stuff regardless of site, right? Why is it being fetched and cached once per site?